PowerEditor/src/Notepad_plus.cpp

```diff
@@ -1089,217 +1089,217 @@ bool Notepad_plus::matchInList(const TCHAR *fileName, const vector<generic_strin
-                        i += counter;
-                        column = tabStop + 1;
-                        tabStop += tabWidth;
-                        counter = 0;
-                        if (i <= currentPos)
-                            ++newCurrentPos;
-                        continue;
-                    }
-                }
-                
-                if (onlyLeading == true && nonSpaceFound == false)
-                    nonSpaceFound = true;
-
-                if (source[i] == '\n' || source[i] == '\r')
-                {
-                    *dest++ = source[i]; 
-                    column = 0;
-                    tabStop = tabWidth - 1;
-                    nonSpaceFound = false;
-                }
-                else if (source[i] == '\t')
-                {
-                    *dest++ = source[i];
-                    column = tabStop + 1;
-                    tabStop += tabWidth;
-                    counter = 0;
-                }
-                else
-                {
-                    *dest++ = source[i];
-                    counter = 0;
-                    if ((source[i] & 0xC0) != 0x80)   // UTF_8 support: count only bytes that don't start with 10......
-                    {
-                        ++column;
-
-                        if (column > 0 && column % tabWidth == 0)
-                            tabStop += tabWidth;
-                    }
-                }
-                
-                if (i <= currentPos)
-                    ++newCurrentPos;
-            }
-            *dest = '\0';
-            break;
-        }
-    }
-
-    _pEditView->execute(SCI_BEGINUNDOACTION);    
-    _pEditView->execute(SCI_SETTEXT, 0, (LPARAM)destination);
-    _pEditView->execute(SCI_GOTOPOS, newCurrentPos);
-
-    for (size_t i=0; i<bookmarks.size(); ++i)
-        _pEditView->execute(SCI_MARKERADD, bookmarks[i], MARK_BOOKMARK);
-
-    for (size_t i=0; i<folding.size(); ++i)
-        _pEditView->fold(folding[i], false);
-
-    _pEditView->execute(SCI_ENDUNDOACTION);
-
-    // clean up
-    delete [] source;
-    delete [] destination;
+	int tabWidth = _pEditView->execute(SCI_GETTABWIDTH);
+	int currentPos = _pEditView->execute(SCI_GETCURRENTPOS);
+	int lastLine = _pEditView->lastZeroBasedLineNumber();
+	int docLength = int(_pEditView->execute(SCI_GETLENGTH)) + 1;
+	if (docLength < 2) 
+		return;
+
+	int count = 0;
+	int column = 0;
+	int counter = 0;
+	int newCurrentPos = 0;
+	int tabStop = tabWidth - 1;   // remember, counting from zero !
+	bool onlyLeading = false;
+	bool nonSpaceFound = false;
+	vector<int> bookmarks;
+	vector<int> folding;
+
+	for (int i=0; i<lastLine; ++i)
+	{
+		if (bookmarkPresent(i))
+			bookmarks.push_back(i);
+
+		if ((_pEditView->execute(SCI_GETFOLDLEVEL, i) & SC_FOLDLEVELHEADERFLAG))
+			if (_pEditView->execute(SCI_GETFOLDEXPANDED, i) == 0)
+				folding.push_back(i);
+	}
+
+	char * source = new char[docLength];
+	if (source == NULL)
+		return;

```

Comment:
Yeah, I was just wondering if you could shed some light on it. Don't take it personally.